drivers/flash/flash_nxp_s32_qspi.h

```diff
@@ -0,0 +1,112 @@
+/*
+ * Copyright 2023-2024 NXP
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+#ifndef ZEPHYR_DRIVERS_FLASH_NXP_S32_QSPI_H_
+#define ZEPHYR_DRIVERS_FLASH_NXP_S32_QSPI_H_
+
+#include "jesd216.h"
+
+#define QSPI_ERASE_VALUE      0xff
+#define QSPI_WRITE_BLOCK_SIZE 1U
```

Comment:
This needs to come from dts (the write block size). Does your driver switch the commands it sends depending upon what some uses? E.g. if I use flash_write() with 129 bytes it will send 32 commands to write 4 bytes followed by 1 command to write 1 byte or will it just fail at the last part? Likewise what if there is a flash memory used with this that does not support 1 byte writes and is configures for 4 byte writes only?

Reply:
This driver will use  `memory-alignment` dt properties (https://github.com/zephyrproject-rtos/zephyr/blob/main/dts/bindings/mtd/nxp%2Cs32-qspi-device.yaml#L17) to pad (by adding 0xFF) at the unaligned data parts. E.g with `memory-alignment=2` use `flash_write()` with 129 bytes it will send 130 bytes (last byte is padded if offset is even number). From application pov, `flash_write()` can be used with any size between 1 and the value specified in `max-program-buffer-size` dt properties.

Reply:
But that is wrong? That's fine if you are using a flash chip, but you get non-flash SPI/QSPI devices (e.g. FRAM) and writing 0xff will write 0xff over data you do not want to touch

Reply:
This driver supports hyperflash chip (using s26hs512 flash chip on board), writing 0xff not affect the existed data.

Reply:
And will completely fail with e.g. https://www.infineon.com/cms/en/product/memories/f-ram-ferroelectric-ram/excelon-f-ram/cy15v104qsn-108sxit/

Reply:
> This driver supports hyperflash chip (using s26hs512 flash chip on board), writing 0xff not affect the existed data.

No hidden writes. Require users to align size and offset to value you are writing. Otherwise you differ from other drivers, at Flash Api level , from other drivers.

Reply:
I added the property dts `write-block-size` for resolve this.

Reply:
People that will write code for subsystem to work on any device have to align offset and size to write-block-size anyway, so by not requiring that. So if somebody delivers something for your device with assumption that data will be padded, will have to fix that when testing code against other device.

Reply:
@de-nordic @nordicjm i believe your concern for this topic has been addressed (by updating the PR to use `write-block-size`), could both of you help me to take another look.

Reply:
> @de-nordic @nordicjm i believe your concern for this topic has been addressed (by updating the PR to use `write-block-size`), could both of you help me to take another look.

You have added a write block size of 1, so does that mean then when I call this function, your driver **writes exactly one byte with no padding or 0xff's what so ever**?

Reply:
you are saying [this write block size](https://github.com/zephyrproject-rtos/zephyr/pull/78073/files#diff-49b9c21ea3c86864ed5469210d04a68afb14267f2b59ff4f5711648899c94d7fR348). this is write block size for a serial NOR flash device. it allows to write exactly 1 byte without padding.

Reply:
I don't mean for a serial NOR flash device, I mean the actual command that comes out the chip, does it send a command to the other device consisting of writing 1 byte and 1 byte only without **any** padding or 0xff's or does it send a command with additional data/padding?

Reply:
After I changed to use `write-block-size`, driver sends command with only data without padding.

Reply:
After I changed to use `write-block-size`, driver send command with only data without padding.

Reply:
Thanks